### PR TITLE
Validator v4 compatibility 

### PR DESCRIPTION
**Pull request by ruudboon**
Hello!

* Type: bug fix
* Link to issue:

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/incubator/blob/master/CONTRIBUTING.md)?
- [ ] I have checked that another pull request for this purpose does not exist.
- [ ] I wrote some tests for this PR.

Small description of change:

- Upgraded Phalcon dependency to v4
- Set > PHP 7.2 requirement
- Fixed return value for validator to fix Declaration errors

Thanks

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,9 +15,7 @@ sudo: false
 dist: trusty
 
 php:
-  - 5.6
-  - 7.0
-  - 7.1
+  - 7.3
```
Comment (Jeckerson) [inline .travis.yml]: Can you move 7.3 below 7.2?

Comment (ruudboon) [reply .travis.yml]: Sure! I moved it up since this is also the setting in the cphalcon.

**Comment by ruudboon** [comment]
I'm seeing some issues with the error messages.
```
Argument 4 passed to Phalcon\Messages\Message::__construct() must be of the type int, null given
```
PrepareCode() returns NULL or an int. Need to investigate if we need to fix this in Phalcon or in the validator.

**Comment by ruudboon** [comment]
Closing this particular pull. Will try to fix all code and make tests work

**Pull request closed without merging**